File: persistence/src/main/java/io/sparkled/persistence/stage/StagePersistenceService.java
package io.sparkled.persistence.stage;

import io.sparkled.model.entity.Stage;
import io.sparkled.model.entity.StageProp;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface StagePersistenceService {

    Stage createStage(Stage stage);

    List<Stage> getAllStages();

    Optional<Stage> getStageById(int stageId);

    List<StageProp> getStagePropsByStageId(int stageId);

    Optional<StageProp> getStagePropByUuid(int stageId, UUID uuid);

    void saveStage(Stage stage, List<StageProp> stageProps);

    void deleteStage(int stageId);
}

File: persistence/src/main/kotlin/io/sparkled/persistence/stage/StagePersistenceService.kt
package io.sparkled.persistence.stage

import io.sparkled.model.entity.Stage
import io.sparkled.model.entity.StageProp
import java.util.UUID

interface StagePersistenceService {

    fun createStage(stage: Stage): Stage

    fun getAllStages(): List<Stage>

    fun getStageById(stageId: Int): Stage?

    fun getStagePropsByStageId(stageId: Int): List<StageProp>

    fun getStagePropByUuid(stageId: Int, uuid: UUID): StageProp?

    fun saveStage(stage: Stage, stageProps: List<StageProp>)

    fun deleteStage(stageId: Int)
}
